src/main/java/uk/gov/companieshouse/api/accounts/service/impl/CompanyAccountServiceImpl.java

```diff
@@ -2,47 +2,36 @@
 
 import java.util.HashMap;
 import java.util.Map;
-import javax.annotation.Resource;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.data.mongodb.repository.MongoRepository;
 import org.springframework.stereotype.Service;
-import uk.gov.companieshouse.GenerateEtagUtil;
 import uk.gov.companieshouse.api.accounts.Kind;
 import uk.gov.companieshouse.api.accounts.LinkType;
 import uk.gov.companieshouse.api.accounts.model.entity.CompanyAccountEntity;
 import uk.gov.companieshouse.api.accounts.model.rest.CompanyAccount;
-import uk.gov.companieshouse.api.accounts.repository.CompanyAccountRepository;
 import uk.gov.companieshouse.api.accounts.service.CompanyAccountService;
-import uk.gov.companieshouse.api.accounts.transformer.CompanyAccountTransformer;
+import uk.gov.companieshouse.api.accounts.transformer.GenericTransformer;
 
 @Service
-public class CompanyAccountServiceImpl implements CompanyAccountService {
+public class CompanyAccountServiceImpl extends AbstractServiceImpl<CompanyAccount, CompanyAccountEntity> implements CompanyAccountService  {
```

Comment:
CompanyAccountServiceImpl implementing CompanyAccountService and extending a AbstractServiceImpl which implements a AbstractService feels a bit overkill. Why are so many interfaces/abstract classes needed?

Reply:
I wanted to add some generic method interfaces that every Service should implement such as addLinks, addEtag, addKind etc so I created the AbstractService that every service should extend from. I then realised that for some methods like addEtag every implementation would be the same so I added the AbstractServiceImpl to provide them.

Reply:
Can't you use one abstract class or an interface with default methods if every service must implement addLinks, etc?

What is the CompanyAccountService interface adding? Why can't you simplify to:
```
public class CompanyAccountService extends AbstractService<CompanyAccount, CompanyAccountEntity>{
```